framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl.java

```diff
@@ -59,49 +57,35 @@ protected void deployCustomResource() {
         logger.info("Creating application from " + OpenShiftResource.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getResourceUrl().toString());
         OpenShiftBinaryClient.getInstance().executeCommand("Deployment failed.", "create", "-f", OpenShiftResource.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getResourceUrl().toString());
 
-        // Wait until deployment is created
-        // Temporary, replace by something better.
+        workbenchRuntimeDeployment = createWorkbenchRuntimeDeployment(project);
+        smartRouterDeployment = createSmartRouterDeployment(project);
+        kieServerDeployment = createKieServerDeployment(project);
+        databaseDeployment = createDatabaseDeployment(project);
+
+        logger.info("Waiting until all services are created.");
         try {
-            new SimpleWaiter(() -> project.getOpenShiftUtil().getServices().size() == 9).timeout(TimeUnit.MINUTES, 1).execute();
+            new SimpleWaiter(() -> workbenchRuntimeDeployment.isReady()).timeout(TimeUnit.MINUTES, 1).execute();
+            new SimpleWaiter(() -> smartRouterDeployment.isReady()).timeout(TimeUnit.MINUTES, 1).execute();
+            new SimpleWaiter(() -> kieServerDeployment.isReady()).timeout(TimeUnit.MINUTES, 1).execute();
+            new SimpleWaiter(() -> databaseDeployment.isReady()).timeout(TimeUnit.MINUTES, 1).execute();
```

Comment:
Shouldn't we check if DB is ready before the Kie Server deployment? E.g. in case when external DB used.

Reply:
This check makes sure that the service is created, there is no need to keep an order.